src/DGtal/shapes/SurfaceMesh.h

```diff
@@ -0,0 +1,773 @@
+/**
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Lesser General Public License as
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ **/
+
+#pragma once
+
+/**
+ * @file SurfaceMesh.h
+ * @author Jacques-Olivier Lachaud (\c jacques-olivier.lachaud@univ-savoie.fr )
+ * Laboratory of Mathematics (CNRS, UMR 5127), University of Savoie, France
+ *
+ * @date 2020/07/11
+ *
+ * Header file for module SurfaceMesh.cpp
+ *
+ * This file is part of the DGtal library.
+ */
+
+#if defined(SurfaceMesh_RECURSES)
+#error Recursive header files inclusion detected in SurfaceMesh.h
+#else // defined(SurfaceMesh_RECURSES)
+/** Prevents recursive inclusion of headers. */
+#define SurfaceMesh_RECURSES
+
+#if !defined SurfaceMesh_h
+/** Prevents repeated inclusion of headers. */
+#define SurfaceMesh_h
+
+//////////////////////////////////////////////////////////////////////////////
+// Inclusions
+#include <iostream>
+#include <sstream>
+#include <string>
+#include "DGtal/base/Common.h"
+#include "DGtal/base/IntegerSequenceIterator.h"
+#include "DGtal/helpers/StdDefs.h"
+
+namespace DGtal
+{
+  /////////////////////////////////////////////////////////////////////////////
+  // template class SurfaceMesh
+  /**
+     Description of template class 'SurfaceMesh' <p> \brief Aim:
+     Represents a mesh as faces and a list of vertices. Vertices may
+     be shared among faces but no correct topology is
+     required. However, you also have methods to navigate between
+     neighbor vertices, faces, etc. The mesh can be equipped with
+     normals at faces and/or vertices.
+
+     We sum up below the possible classes for representing meshes:
```

Comment:
Maybe this should be copied (or moved elsewhere). 
Eg:
 
* on packageShapes create a new subitem level grouping "Manipulating Simple Mesh (Bertrand Kerautret)", 
"Half-edge data structure, triangulated surfaces and polygonal surfaces (Jacques-Olivier Lachaud)" and "Surface mesh data structure for representing manifold or non-manifold polygonal surfaces in R3 (Jacques-Olivier Lachaud)
*  Just copie this paragraph to a very short module page heading the new subitems list

Do you see what I mean ? I have the feeling that this list should be top level and preliminary to the dedicated module page

Reply:
Agreed. I will make a proposition

Reply:
Done